lib/compose.js

```diff
@@ -50,10 +46,39 @@ function asyncify(f) {
   }
 }
 
+// This accepts an array of arrays and outputs a combination of their layers.
+//
```

Comment:
had a look and found : 
- https://www.npmjs.com/package/js-combinatorics#cartesian-product
- https://www.npmjs.com/package/cartesian-product  <<< this one does just one thing

Reply:
Thanks! Just added that module, so it's doing the heavy lifting for us now. :muscle: